test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +78,80 @@
 (expect false (values-are-valid-emails? [100]))
 (expect false (values-are-valid-emails? ["true"]))
 (expect false (values-are-valid-emails? ["false"]))
+
+;; Tests to avoid analyzing hidden tables
+(defn count-un-analyzed-fields [db_id table-name]
```

Comment:
For the sake of making refactoring easier I think it makes sense to keep functions like these private if they're not being used in other namespaces. If we wanted to change this function or even remove it and it was `defn-` it would save us from having to grep the codebase for other uses